src/test/java/com/angelbroking/smartapi/SmartConnectTest.java

```diff
@@ -0,0 +1,176 @@
+package com.angelbroking.smartapi;
+
+import com.angelbroking.smartapi.http.SmartAPIRequestHandler;
+import com.angelbroking.smartapi.http.exceptions.SmartAPIException;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.IOException;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(MockitoJUnitRunner.Silent.class)
+@Slf4j
+public class SmartConnectTest {
+
+    @Mock
+    private SmartAPIRequestHandler smartAPIRequestHandler;
+
+    @Mock
+    private SmartConnect smartConnect;
+
+    @Mock
+    private Routes routes;
+    private String apiKey;
+    private String accessToken;
+
+
+    @Before
+    public void setup() {
+         apiKey = System.getProperty("apiKey");
+         accessToken = System.getenv("accessToken");
+    }
+
+
+    @Test
+    public void testMarketData_Success() throws SmartAPIException, IOException {
+        String url = routes.get("api.market.data");
+        JSONObject params = getMarketDataRequest();
+        when(smartAPIRequestHandler.postRequest(eq(this.apiKey), eq(url), eq(params), eq(this.accessToken))).thenReturn(createMarketDataResponse());
+        try{
+            JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, this.accessToken);
+             response.getJSONObject("data");
+        }catch (Exception | SmartAPIException e) {
+            e.printStackTrace();
+            System.out.println(e.getMessage());
```

Comment:
System.out.println(e.getMessage());?

Reply:
I Have taken references from other methods, In the catch block the error message is printed.
I can use logger instead of System.out.println and handle null but the code will differ from other methods.